math/src/field/smallprimefield/generic_prime_field_elt.rs

```diff
@@ -0,0 +1,426 @@
+use crate::field::{
+    traits::{FieldElement, StarkField},
+    QuadExtensionA,
+};
+use core::{
+    convert::{TryFrom, TryInto},
+    fmt::{Debug, Display, Formatter},
+    mem,
+    ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Neg, Sub, SubAssign},
+    slice,
+};
+use utils::{
+    collections::Vec,
+    string::{String, ToString},
+    AsBytes, ByteReader, ByteWriter, Deserializable, DeserializationError, Randomizable,
+    Serializable,
+};
+
+// Number of bytes needed to represent field element
+const ELEMENT_BYTES: usize = mem::size_of::<u64>();
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Default)]
+pub struct GenericPrimeFieldElt<const M: u64, const G: u64> {
+    pub(crate) value: u64,
+}
+
+impl<const M: u64, const G: u64> GenericPrimeFieldElt<M, G> {
+    // TODO: this may need to be something else, e.g. its own type param
+    const fn get_twoadic_root() -> Self {
+        Self::new(G)
+    }
+
+    const fn get_modulus_bits() -> u32 {
+        64u32 - M.leading_zeros()
+    }
+
+    /// Creates a new field element from a u128 value. If the value is greater than or equal to
+    /// the field modulus, modular reduction is silently preformed. This function can also be used
+    /// to initialize constants.
+    pub const fn new(value: u64) -> Self {
+        let val = value % M; //if  == Ordering::Less { value } else { };
+        Self { value: val }
+    }
+
+    /// Returns field element converted to u64 representation.
+    pub fn as_u64(&self) -> u64 {
+        self.value
+    }
+
+    #[allow(unused)]
+    unsafe fn get_power_series(b: Self, n: usize) -> Vec<Self> {
+        let mut result = utils::uninit_vector(n);
+        result[0] = Self::ONE;
+        for i in 1..result.len() {
+            result[i] = result[i - 1] * b;
+        }
+        result
+    }
+}
```

Comment:
Can we use `get_power_series()` function from the `utils` module instead of this one? If so, we can probably delete this function.